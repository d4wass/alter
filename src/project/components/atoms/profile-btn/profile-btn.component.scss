@use '../../../../base';

//TODO: move to some different stylesheet file for defaults styles
.btn-default {
  padding: 10px;
  background-color: #e9ecef;
  border-radius: 5px;
  border: 1px solid #ced4da;
  font-weight: bold;
  color: #212529;
  font-family: 'Roboto', 'Helvetica Neue', sans-serif;
  cursor: pointer;
  transition: all ease-in-out 0.3s;

  &:hover {
    background-color: base.$yellow-span;
    border: 1px solid base.$yellow-span;
  }
}
