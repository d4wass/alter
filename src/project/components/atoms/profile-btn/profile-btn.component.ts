import { Component, Input, Output, EventEmitter, ChangeDetectionStrategy } from '@angular/core';

@Component({
  selector: 'app-profile-btn',
  template: `
    <!-- <ng-container *ngIf="btnType !== 'cancel'">
      <button (click)="handleProfileEdit()">
        <p *ngIf="isEditView">Save Profile</p>
        <p *ngIf="!isEditView">Edit Profile</p>
      </button>
    </ng-container>
    <ng-container *ngIf="btnType === 'cancel' && isEditView">
      <button (click)="handleProfileCancel()">
        <p>Cancel</p>
      </button>
    </ng-container> -->
    <ng-container>
      <button (click)="handleClick()"><ng-content></ng-content></button>
    </ng-container>
  `,
  styleUrls: ['./profile-btn.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class ProfileBtnComponent {
  @Output() handleClickEvent: EventEmitter<any> = new EventEmitter<any>();

  handleClick() {
    this.handleClickEvent.emit();
  }
}
