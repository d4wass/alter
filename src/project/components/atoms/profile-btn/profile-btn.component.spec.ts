import { createComponentFactory, Spectator } from '@ngneat/spectator';
import { ProfileBtnComponent } from './profile-btn.component';

describe('ProfileBtnComponent', () => {
  let component: ProfileBtnComponent;
  let spectator: Spectator<ProfileBtnComponent>;

  const createComponent = createComponentFactory({
    component: ProfileBtnComponent
  });

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
  });

  it('should match to snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  describe('@Outputs', () => {
    it('should emit (handleClickEvent) on btn click event', () => {
      const spy = jest.spyOn(component, 'handleClick');
      spectator.dispatchMouseEvent('.btn-default', 'click');

      expect(spy).toHaveBeenCalled();
    });
    it('should emit (handleClickEvent) on handleClick() method call', () => {
      const spy = jest.spyOn(component.handleClickEvent, 'emit');
      component.handleClick();

      expect(spy).toHaveBeenCalled();
    });
  });
});
