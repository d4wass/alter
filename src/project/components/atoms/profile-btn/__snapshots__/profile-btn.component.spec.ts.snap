// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileBtnComponent should match to snapshot 1`] = `
<app-profile-btn
  handleClickEvent={[Function EventEmitter_]}
>
  <button
    class="btn-default"
    data-test="btn-default"
  />
</app-profile-btn>
`;
